bundles/org.smarthomej.transform.basicprofiles/src/main/java/org/smarthomej/transform/basicprofiles/internal/profiles/StateFilterProfile.java

```diff
@@ -0,0 +1,226 @@
+            }
+
+            return parsedConditions;
+        } catch (IllegalArgumentException e) {
+            logger.warn("Cannot parse condition {}. Expected format ITEM_NAME <EQ|NEQ> STATE_VALUE: '{}'", config,
+                    e.getMessage());
+            return List.of();
+        }
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return STATE_FILTER_UID;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State state) {
+        // do nothing
+    }
+
+    @Override
+    public void onCommandFromItem(Command command) {
+        callback.handleCommand(command);
+    }
+
+    @Override
+    public void onCommandFromHandler(Command command) {
+        callback.sendCommand(command);
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State state) {
+        State resultState = checkCondition(state);
+        if (resultState != null) {
+            logger.debug("Received state update from handler: {}, forwarded as {}", state, resultState);
+            callback.sendUpdate(resultState);
+        } else {
+            logger.debug("Received state update from handler: {}, not forwarded to item", state);
+        }
+    }
+
+    @Nullable
+    private State checkCondition(State state) {
+        if (!conditions.isEmpty()) {
+            boolean allConditionsMet = true;
+            for (Condition condition : conditions) {
+                logger.debug("Evaluting condition: {}", condition);
+                try {
+                    Item item = itemRegistry.getItem(condition.itemName);
+                    String itemState = item.getState().toString();
+
+                    if (!condition.matches(itemState)) {
+                        allConditionsMet = false;
+                    }
+                } catch (ItemNotFoundException e) {
+                    logger.warn(
+                            "Cannot find item '{}' in registry - check your condition expression - skipping state update",
+                            condition.itemName);
+                    allConditionsMet = false;
+                }
+
+            }
+            if (allConditionsMet) {
+                return state;
+            } else {
+                return configMismatchState;
+            }
+        } else {
+            logger.warn(
+                    "No configuration defined for StateFilterProfile (check for log messages when instantiating profile) - skipping state update");
+        }
+
+        return null;
+    }
+
+    @Nullable
+    State parseState(@Nullable String stateString) {
+        // Quoted strings are parsed as StringType
+        if (stateString == null) {
+            return null;
+        } else if (stateString.startsWith("'") && stateString.endsWith("'")) {
+            return new StringType(stateString.substring(1, stateString.length() - 1));
+        } else {
+            return TypeParser.parseState(acceptedDataTypes, stateString);
+        }
+    }
+
+    class Condition {
```

Comment:
```suggestion
    class StateCondition {
```

and move to a separate file